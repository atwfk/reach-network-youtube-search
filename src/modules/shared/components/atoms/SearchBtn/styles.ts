import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledSearchButton = styled.button`
  border: none;
  border-left: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;

  @media (min-width: ${SCREENS.MD}) {
    border: 1px solid #9d9d9d;
    padding: 0px 25px;
    opacity: ${(props) => props.disabled && 0.6};
    background-color: #eee;
  }

  .search-icon {
    color: #eee;
    font-size: 20px;
    opacity: ${(props) => props.disabled && 0.6};
    margin-left: 10px;

    @media (min-width: ${SCREENS.MD}) {
      color: #9d9d9d;
      margin: 0;
    }
  }
`;
