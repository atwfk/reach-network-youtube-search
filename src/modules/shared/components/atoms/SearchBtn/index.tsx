import React from "react";
import type { FC, ReactElement } from "react";
import { AiOutlineSearch } from "react-icons/ai";
import { StyledSearchButton } from "./styles";

const SearchBtn: FC<{ type: "button" | "submit" | "reset" | undefined; disabled?: boolean }> = ({
  type,
  disabled
}): ReactElement => {
  return (
    <StyledSearchButton type={type} disabled={disabled} data-testid="search-btn">
      <AiOutlineSearch className="search-icon" />
    </StyledSearchButton>
  );
};

export default SearchBtn;
