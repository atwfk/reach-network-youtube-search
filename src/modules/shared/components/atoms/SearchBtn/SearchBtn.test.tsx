import { render, screen, cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import SearchBtn from ".";

describe("SearchBtn Component snapshots", () => {
  it("Should render clickable SearchBtn", () => {
    const tree = renderer.create(<SearchBtn type="button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(cleanup);

describe("SearchBtn Component RTL", () => {
  it("should have disabled search button when we pass {disabled}", () => {
    render(<SearchBtn type="button" disabled />);

    const searchBtn = screen.getByTestId("search-btn");

    expect(searchBtn).toBeDisabled();
  });

  it("should not have disabled search button when we don't pass {disabled}", () => {
    render(<SearchBtn type="button" />);

    const searchBtn = screen.getByTestId("search-btn");

    expect(searchBtn).not.toBeDisabled();
  });
});
