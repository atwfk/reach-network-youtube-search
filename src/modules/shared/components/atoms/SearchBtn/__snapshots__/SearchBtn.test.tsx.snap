// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBtn Component snapshots Should render clickable SearchBtn 1`] = `
.c0 {
  border: none;
  border-left: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
}

.c0 .search-icon {
  color: #eee;
  font-size: 20px;
  margin-left: 10px;
}

@media (min-width:768px) {
  .c0 {
    border: 1px solid #9d9d9d;
    padding: 0px 25px;
    background-color: #eee;
  }
}

@media (min-width:768px) {
  .c0 .search-icon {
    color: #9d9d9d;
    margin: 0;
  }
}

<button
  className="c0"
  data-testid="search-btn"
  type="button"
>
  <svg
    className="search-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
    />
  </svg>
</button>
`;
